framework_tvb/tvb/core/adapters/abcadapter.py

```diff
@@ -448,28 +449,28 @@ def _is_group_launch(self):
     def _get_output_path(self):
         return self.storage_path
 
-    @staticmethod
-    def load_entity_by_gid(data_gid):
+    def load_entity_by_gid(self, data_gid):
+        # type: (typing.Union[uuid.UUID, str]) -> DataType
         """
         Load a generic DataType, specified by GID.
         """
-        if isinstance(data_gid, uuid.UUID):
-            data_gid = data_gid.hex
-        return load_entity_by_gid(data_gid)
+        idx = load_entity_by_gid(data_gid)
+        if idx and self.generic_attributes.parent_burst is None:
+            # Only in case the BurstConfiguration references hasn't been set already, take it from the current DT
+            self.generic_attributes.parent_burst = idx.fk_parent_burst
+        return idx
```

Comment:
This method changed is the main part of the solution.
The other changes are related refactoring (some strictly necessary for the staticmethod removal). See the commits order for some more details.